### DISCUSSION TITLE
Performance - PHP Transaction time

### DISCUSSION THREAD

**Discussion by brnpimentel** [shinsenter/php#25] [General]
Hello, I am using the `phpfpm-apache:php7.4-tidy` image in production but I am experiencing some performance problems. 

I have another legacy infrastructure that is configured in a simple way apache + php fpm, without using docker/kubernetes. 

In both are installed New Relic APM, to application monitor. What I can not understand is the poor performance using 
kubernetes. And I don't even mean the overhead to kubernetes, but the transaction time of PHP. Using docker with the image I just mentioned is about **60% slower.** 

I tried to make some configurations but without success to improve these numbers.

Is anyone facing the same problem? Any suggestions?

```
ENV ENABLE_CRONTAB=false
ENV PHP_DISPLAY_ERRORS="On"
ENV PHP_ERROR_REPORTING="E_ALL"
ENV PHP_MEMORY_LIMIT="512M"
ENV PHP_MAX_EXECUTION_TIME="90"
ENV PHP_UPLOAD_MAX_FILE_SIZE="8M"
ENV PHP_PM_CONTROL=dynamic
ENV PHP_PM_MAX_CHILDREN=12
ENV PHP_PM_MAX_SPARE_SERVERS=6
ENV PHP_PM_MIN_SPARE_SERVERS=3
ENV PHP_PM_START_SERVERS=6
```

**Comment by shinsenter** [comment]
Hey @brnpimentel 

The performance of an application running inside a Docker container may not be optimal due to various reasons. For example, the speed of accessing files from the host machine to the Docker container can affect the overall performance. If the host machine's disk speed is slow, then the container's I/O (input/output) performance would also be slow. This can lead to the application running slower than expected.

Another reason for slow performance could be related to the network connectivity between the Docker container and the host machine. If there is a slow or unreliable network connection, it can result in slow performance for the application running inside the container. For example, if the application relies on accessing a database that is located on a different machine, the network connection speed between the container and the database server can affect the overall performance.

I don't know the exact configurations you have tried, so it's difficult for me to diagnose your specific situation. Could you provide me with more information about your host machine and the way you have set up Docker?
